mainnet-rehearsals/3-R4-upgrade-2-of-2/README.md

```diff
@@ -0,0 +1,174 @@
+    (0xce479c1299adbe12f34a479f404937151af19d92) is changed to
+    0x3311aC7F72bb4108d9f4D5d50E7623B1498A9eC0.
+2. The implementation (storage key is the same as above) of the
+   `OptimismPortalProxy` contract
+   (0xDd3840F548Bd8fA41421050ed7C28753B494734c) is changed to
+   0x0A9d47e531825FaaA2863D4d10DC8E5E0B91BfB0 ([source
+   code](https://github.com/ethereum-optimism/optimism/blob/716f81a6fc4ef125364b95a799474082ea3eb062/packages/contracts-bedrock/src/L1/OptimismPortal.sol#L123)).
+3. The `OptimismPortalProxy` contract is being reinitialized with
+    1. storage slot `0x0` (_initialized) set to `0x2`.
+    2. storage slot `0x1` (ResourceParams) set to
+       `0x000000000117814800000000000000000000000000000000000000003b9aca00`,
+       which means prevBlockNum is set to 0x1178148, prevBoughtGas is
+       set to 0, and prevBaseFee is set to 0x3b9aca00.
+    3. storage slot `0x32` (l2Sender) set to `0xdead`.
+    4. storage slot `0x35` (l2Oracle) set to `0x0000000000000000000000dfe97868233d1aa22e815a266982f2cf17685a2700`.
+    5. storage slot `0x36` (systemConfig) set to `0x000000000000000000000000229047fed2591dbec1ef1118d64f7af3db9eb290`.
+    6. storage slot `0x37` (guardian) set to `0x0000000000000000000000009ba6e03d8b90de867373db8cf1a58d2f7f006b3a`.
+
+
+All of these addresses should be part of the Optimism Governance vote
+that approves this upgrade if this is a [Normal
+Operation](https://github.com/ethereum-optimism/OPerating-manual/blob/1f42a3766d084864a818b93ce7ba0857a4a846ea/Security%20Council%20Charter%20v0.1.md#normal-operation).
+
+![](./images/tenderly-state-diff.png)
+
+
+#### 3.3. Validate and extract domain hash and message hash to approve.
+
+Now that we have verified the transaction performs the right
+operation, we need to validate and extract domain hash and message
+hash to approve.
+
+Go back to the "Overview" tab, and find the first
+`GnosisSafe.domainSeparator` call. This call's return value will be
+the domain hash that will show up in your Ledger:
+`0xdbf84cb7cfac409dddf2f9e1e3dec451e4fb9a7abe42d83ef279baff12b1d730`.
+
+Right before this `GnosisSafe.domainSeparator` call, you will see a
+call to `GnosisSafe.encodeTransactionData`. Its return value will be a
+concatnation of `0x1901`, the domain hash, and the message hash:
+`0x1901[domain hash][message hash]`, which is
+`0x1901dbf84cb7cfac409dddf2f9e1e3dec451e4fb9a7abe42d83ef279baff12b1d730d042638edac0888e1c73d74fb5e53185278005b41bd8d43804d0423cce0f0ea1`
+in this case.
+
+![](./images/tenderly-hashes-1.png)
+
+![](./images/tenderly-hashes-2.png)
+
+### 4. Approve the signature on your ledger
+
+Once the validations are done, it's time to actually sign the
+transaction. Make sure your ledger is still unlocked and run the
+following:
+
+``` shell
+just sign-council # or just sign-council <hdPath>
+```
+
+!!! IMPORTANT: This is the most security critical part of the
+playbook: make sure the domain hash and message hash in the following
+two places match:
```

Comment:
I think this is meant to appear as a special callout, but it doesn't seem to work in github. I only noticed this because my IDE rendered it, but broke on the intermediate `:`.